core/src/main/java/org/apache/calcite/plan/SubstitutionVisitor.java

```diff
@@ -1717,6 +1907,31 @@ private static UnifyResult setOpApply(UnifyRuleCall call) {
     return null;
   }
 
+  /**
+   * Check if query equivalent to target required columns, and also check correlationId
+   */
+  private static boolean checkRequiredColumnsAndCorrelationId(MutableCorrelate query,
+      MutableCorrelate target) {
+    if (!(query.correlationId.getId() == target.correlationId.getId())) {
```

Comment:
What if query and materialized view are equivalent with each other by semantics but only different by the correlationId. Should we fail the match ?

Reply:
I think we should check that it's exactly the same，Maybe you can give an example. Let's discuss it together.